libnmstate/netapplier.py

```diff
@@ -167,6 +169,44 @@ def _apply_ifaces_state(
         raise
 
 
+def _prepare_vlan_ifaces_for_mtu_change(desired_state, current_state):
+    current_ifaces = current_state.interfaces
+    desired_ifaces = desired_state.interfaces
+    desired_ifaces_names = desired_ifaces.keys()
+    current_changed_vlan_ifaces = [
+        (ifname, ifstate)
+        for ifname, ifstate in current_ifaces.items()
+        if _is_vlan_iface(ifstate)
```

Comment:
These conditions will always resolve to false, which tells me that there are no tests to cover this logic.
The last two conditions are contradicting.

Reply:
Why? I have debugged it and this is not always false.

Reply:
If ifname is not in the desired state (cond 3) it cannot be in UP state (cond 2).

Reply:
I think I am missing something here. Do you mean current state? Why all the UP interfaces must be in the desired state?

Reply:
Please explain me the scenarios in which the last two lines below will not give False.

```
desired_ifaces_names = desired_ifaces.keys()

ifname = ‘eth1’ # it is in the current state

 _is_up_iface(desired_ifaces.get(ifname, {}))

and ifname not in desired_ifaces_names

```

Reply:
You are right. I've done some changes. Please take a look, the current solution proposed by @cathay4t is now working. I have added another integration test to check what happens if the VLANs are not set in the desired state and it is working.